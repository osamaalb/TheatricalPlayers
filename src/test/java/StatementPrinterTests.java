import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.TEXT);

        verify(result);
    }

    @Test
    void statementWithNewPlayTypes() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("henry-v",  new Play("Henry V", "history"));
        plays.put("as-like",  new Play("As You Like It", "pastoral"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        Assertions.assertThrows(Error.class, () -> {
            invoice.print(plays, Invoice.PrintFormat.TEXT);
        });
    }

    @Test
    void testOtherCases() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.TEXT);

        verify(result);
    }

    @Test
    void exampleStatementHTML() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", "tragedy"));
        plays.put("as-like",  new Play("As You Like It", "comedy"));
        plays.put("othello",  new Play("Othello", "tragedy"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("othello", 40)));

        var result = invoice.print(plays, Invoice.PrintFormat.HTML);

        verify(result);
    }
}
