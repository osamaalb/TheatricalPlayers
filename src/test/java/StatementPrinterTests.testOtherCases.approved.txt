Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $500.00 (40 seats)
Amount owed is $1,260.00
You earned 14 credits
